ipfs-unixfs/src/types.rs

```diff
@@ -68,8 +68,6 @@ impl Block {
         let expected_links = parse_links(&self.cid, &self.data)?;
         let mut actual_links = self.links.clone();
         actual_links.sort();
-        // TODO: why do the actual links need to be deduplicated?
```

Comment:
Good question!
We were passing a set to a parameter named set, and it was named that because the _intent_ of the method was to return to you the _set_ of all the CIDs referenced (as links/children). Not to return the exact of links/children. 
So, to compare the full list to just the list of CIDs referenced, you have to dedup (to match what the BTreeSet did).

I'm arguably being moderately abusive by passing  Vec in as set, but it allows the duplicates (and order!) to be preserved.